.github/workflows/build_and_test.yml

```diff
@@ -629,7 +629,22 @@ jobs:
     - name: Spark connect jvm client mima check
       if: inputs.branch != 'branch-3.3'
       run: ./dev/connect-jvm-client-mima-check
+    - name: Install Python linter dependencies for branch-3.3
+      if: inputs.branch == 'branch-3.3'
+      run: |
+        # SPARK-44554: Copy from https://github.com/apache/spark/blob/073d0b60d31bf68ebacdc005f59b928a5902670f/.github/workflows/build_and_test.yml#L501-L508
+        # Should delete this section after SPARK 3.3 EOL.
+        python3.9 -m pip install 'flake8==3.9.0' pydata_sphinx_theme 'mypy==0.920' 'pytest==7.1.3' 'pytest-mypy-plugins==1.9.3' numpydoc 'jinja2<3.0.0' 'black==21.12b0'
+        python3.9 -m pip install 'pandas-stubs==1.2.0.53'
+    - name: Install Python linter dependencies for branch-3.4
+      if: inputs.branch == 'branch-3.4'
+      run: |
+        # SPARK-44554: Copy from https://github.com/apache/spark/blob/a05c27e85829fe742c1828507a1fd180cdc84b54/.github/workflows/build_and_test.yml#L571-L578
+        # Should delete this section after SPARK 3.4 EOL.
+        python3.9 -m pip install 'flake8==3.9.0' pydata_sphinx_theme 'mypy==0.920' 'pytest==7.1.3' 'pytest-mypy-plugins==1.9.3' numpydoc 'jinja2<3.0.0' 'black==22.6.0'
+        python3.9 -m pip install 'pandas-stubs==1.2.0.53' ipython 'grpcio==1.48.1' 'grpc-stubs==1.24.11' 'googleapis-common-protos-stubs==2.2.0'
     - name: Install Python linter dependencies
+      if: inputs.branch != 'branch-3.3' && inputs.branch != 'branch-3.4'
```

Comment:
cc  @HyukjinKwon FYI

There change to installed different dependencies for different branches. If we think this is too complex, I think we can also skip `Install Python linter dependencies` and  `Python linter` for `branch-3.3/3.4` directly

Reply:
I'm not sure about this, maybe we need dedicated workflows for 3.3/3.4/3.5, instead of reusing the master one

let's wait for more feedbacks

Reply:
cc @Yikun

Reply:
Yes, if we could use `yml` files from different branches for daily testing, there wouldn't be this problem.

Reply:
Another problem we are suffering is the Python version:
after we upgraded the Python version to 3.9, all the workflows only tests against 3.9, then the support of other versions are broken from time to time, e.g. https://github.com/apache/spark/pull/39542, https://github.com/apache/spark/pull/39844, https://github.com/apache/spark/pull/40153

Reply:
I guess it's worthwhile to use one workflow per branch, for example, after we upgrade python to 3.10 or 3.11 in 4.0, we still test branch-3.5 with 3.9.

Reply:
- It seems we can't trigger a period job in a branch.
- branch push job is separate and using separate docker image cahce

Reply:
So should we fix it in the current way?